### PR TITLE
Added tags for efficiency and consolidate SSEs

### PR DESCRIPTION
**Pull request by umbreak**
Fixes https://github.com/BlueBrain/nexus/issues/2094
Fixes https://github.com/BlueBrain/nexus/issues/2095

### CODE REVIEW

**Review by wwajerowicz — APPROVED** [review]

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/Event.scala [hunk 1]
```diff
@@ -34,8 +34,20 @@ object Event {
       * @return the project where the event belongs
       */
     def project: ProjectRef
+
+  }
+
+  trait OrganizationScopedEvent extends Event {
```
Comment (imsdu) [inline delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/Event.scala:40]: We don't really need this for `OrganizationScopedEvent` as only OrganizationEvent fits in it no ?

File: delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/resources/ResourcesImpl.scala [hunk 2]
```diff
@@ -216,13 +216,7 @@ object ResourcesImpl {
...
       next = Resources.next,
       evaluate = Resources.evaluate(resourceResolution),
-      tagger = (ev: ResourceEvent) =>
-        Set(
-          Event.eventTag,
-          moduleType,
-          Projects.projectTag(ev.project),
-          Organizations.orgTag(ev.project.organization)
-        ),
+      tagger = EventTags.forProjectScopedEvent(moduleType),
```
Comment (imsdu) [inline delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/resources/ResourcesImpl.scala:219]: I understand that for light and rare events, we might want to add more tags but for resources, the cost may be a little high to have 6 tags for resources no ?

**Review by imsdu — CHANGES_REQUESTED** [review]

File: delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/resources/ResourcesImpl.scala (hunk 2 above)
Comment (umbreak) [reply delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/resources/ResourcesImpl.scala:219]: We agreed when we discussed that issue that we are going to do it for all events.

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/Event.scala (hunk 1 above)
Comment (umbreak) [reply delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/Event.scala:40]: Does it hurt? I wanted to have it for consistency. Now we have 3 subclasses of `Event` and each of them makes sense.

File: delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/resources/ResourcesImpl.scala (hunk 2 above)
Comment (umbreak) [reply delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/resources/ResourcesImpl.scala:219]: It is precisely for resources and files that this is going to become useful.

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/Event.scala (hunk 1 above)
Comment (imsdu) [reply delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/Event.scala:40]: It felt also weird when I read it as `ProjectScopedEvent` could be seen as `OrganizationScopedEvent`
Wouldn't it make more sense if you want to keep `OrganizationScopedEvent` to have that approach ?
Transform `label` to `organizationLabel` implement it in `ProjectScopedEvent` and in EventTags, `forProjectScopedEvent` would be `forOrganizationScopedEvent` merged with project related tags ?

**Review by imsdu — APPROVED** [review]

**Review by wwajerowicz — APPROVED** [review]

**Pull request merged**